### PR TITLE
Split unit tests out of glean-core/src/lib.rs

### PR DESCRIPTION
**Pull request by Dexterp37**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

**Comment by Dexterp37** [comment]
> 
> 
> Shouldn't the `lib_unit_tests.rs` file go in the `/tests/` directory?

The tests in this file rely on private data from `lib.rs`. I initially moved them to `/tests/` but that is seen as a separate crate and, as such, does not have access to private stuff.

**Comment by badboy** [comment]
> Shouldn't the `lib_unit_tests.rs` file go in the `/tests/` directory?

Stuff in `/tests/` is considered "integration tests" and thus should only test the public API (in fact, that's all it can test because it consumes the library as an external crate, just as your users would).

**Comment by travis79** [comment]
> > Shouldn't the `lib_unit_tests.rs` file go in the `/tests/` directory?
> 
> Stuff in `/tests/` is considered "integration tests" and thus should only test the public API (in fact, that's all it can test because it consumes the library as an external crate, just as your users would).

TIL: Good to know about the scope of the /tests folder!

File: glean-core/src/lib.rs [hunk 1]
```diff
@@ -534,334 +534,6 @@ impl Glean {
...
-    #[test]
-    fn disabling_when_already_disabled_is_a_noop() {
-        let dir = tempfile::tempdir().unwrap();
-        let tmpname = dir.path().display().to_string();
-        let mut glean = Glean::with_options(&tmpname, GLOBAL_APPLICATION_ID, false).unwrap();
-
-        assert!(!glean.set_upload_enabled(false));
-    }
-}
+mod lib_unit_tests;
```
[the remark this answers is not in the document]
Comment (Dexterp37) [reply glean-core/src/lib.rs]: OH! TIL! I like it!

**Pull request merged**